dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/FeedbackRestRepository.java

```diff
@@ -0,0 +1,102 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang.StringUtils;
+import org.dspace.app.rest.exception.DSpaceBadRequestException;
+import org.dspace.app.rest.exception.DSpaceFeedbackNotFoundException;
+import org.dspace.app.rest.exception.RepositoryMethodNotImplementedException;
+import org.dspace.app.rest.exception.UnprocessableEntityException;
+import org.dspace.app.rest.model.FeedbackRest;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.service.FeedbackService;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the Repository that takes care of the operations on the {@link FeedbackRest} objects
+ * 
+ * @author Mykhaylo Boychuk (mykhaylo.boychuk@4science.com)
+ */
+@Component(FeedbackRest.CATEGORY + "." + FeedbackRest.NAME)
+public class FeedbackRestRepository extends DSpaceRestRepository<FeedbackRest, Integer> {
+
+    @Autowired
+    private FeedbackService feedbackService;
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Override
+    @PreAuthorize("hasAuthority('AUTHENTICATED')")
+    public Page<FeedbackRest> findAll(Context context, Pageable pageable) {
+        throw new RepositoryMethodNotImplementedException(FeedbackRest.NAME, "findAll");
+    }
+
+    @Override
+    @PreAuthorize("hasAuthority('AUTHENTICATED')")
+    public FeedbackRest findOne(Context context, Integer id) {
+        throw new RepositoryMethodNotImplementedException(FeedbackRest.NAME, "findOne");
+    }
+
+    @Override
+    @PreAuthorize("permitAll()")
+    protected FeedbackRest createAndReturn(Context context) throws AuthorizeException, SQLException {
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        FeedbackRest feedbackRest = null;
+
+        String recipientEmail = configurationService.getProperty("feedback.recipient");
+        if (StringUtils.isBlank(recipientEmail)) {
+            throw new DSpaceFeedbackNotFoundException("Recipient's email was not found!");
+        }
+
+        try {
+            feedbackRest = mapper.readValue(req.getInputStream(), FeedbackRest.class);
+        } catch (IOException exIO) {
+            throw new UnprocessableEntityException("error parsing the body " + exIO.getMessage(), exIO);
+        }
+
+        String senderEmail = feedbackRest.getEmail();
+        String message = feedbackRest.getMessage();
+
+        if (StringUtils.isBlank(senderEmail) || StringUtils.isBlank(message)) {
+            throw new DSpaceBadRequestException("Filds as e-mail and message are mandatory!");
```

Comment:
Small spelling/grammar errors here. Should say something like "The fields email and message are mandatory!"